### PR TITLE
Change u-root to optionally use gobusybox

### PR DESCRIPTION
**Pull request by rminnich**
This is a notional idea of how we might use to move to the
gobusybox, which works with modules correctly.

The u-root command is widely used, so just dumping it is not really an option.
Asking people to use a new command is not that great either.

Instead, we add an option, gobusybox, which will cause u-root to use the
gobusybox package, and hence work with modules.

This will, in turn, let us start using github actions, which seem to really want
us to use modules.

Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

### CODE REVIEW

File: u-root.go [hunk 1]
```diff
@@ -136,11 +147,74 @@ func generateLabel() string {
...
+		os.RemoveAll(tmpDir)
+	}
+}
+
 func main() {
 	flag.Parse()
 
 	start := time.Now()
 
+	if *usegobusybox {
```
Comment (rjoleary) [inline u-root.go:212]: Can you enable this option in the uroot_test.go file from now on?

File: u-root.go [hunk 2]
```diff
@@ -136,11 +147,74 @@ func generateLabel() string {
 	return fmt.Sprintf("%s-%s-%s-%s", *build, env.GOOS, env.GOARCH, strings.Join(baseCmds, "_"))
 }
 
+func gobusyboxMain() error {
+	bopts := &golang.BuildOpts{}
+	// hmm.
+	//	bopts.RegisterFlags(flag.CommandLine)
+	flag.Parse()
+
+	// Why doesn't the log package export this as a default?
```
Comment (rjoleary) [inline u-root.go:156]: Add TODO

File: u-root.go [hunk 3]
```diff
@@ -136,11 +147,74 @@ func generateLabel() string {
 	return fmt.Sprintf("%s-%s-%s-%s", *build, env.GOOS, env.GOARCH, strings.Join(baseCmds, "_"))
 }
 
+func gobusyboxMain() error {
+	bopts := &golang.BuildOpts{}
+	// hmm.
+	//	bopts.RegisterFlags(flag.CommandLine)
```
Comment (rjoleary) [inline u-root.go:153]: I think you need to add the *tags somewhere in here.

File: u-root.go (hunk 1 above)
Comment (rminnich) [reply u-root.go:209]: one war at a time :-)

File: u-root.go (hunk 3 above)
Comment (rminnich) [reply u-root.go]: hmm. where :-)

File: cmds/exp/bzimage/bzimage_test.go [hunk 4]
```diff
@@ -106,7 +106,8 @@ var (
 `
 )
 
-func TestSimple(t *testing.T) {
+// Turn this off until modules are done.
+func testSimple(t *testing.T) {
```
Comment (rjoleary) [inline cmds/exp/bzimage/bzimage_test.go:110]: How is this possible? testSimple is completely unrelated.

File: pkg/uroot/uroot_test.go [hunk 5]
```diff
@@ -23,10 +23,15 @@ type inMemArchive struct {
 	*cpio.Archive
 }
 
+// if GO111MODULES is not "off", i.e. we are probably using
+// modules, there are tests we need to skip.
+var skipModules = !(os.Getenv("GO111MODULE") == "off")
```
Comment (rjoleary) [inline pkg/uroot/uroot_test.go:28]:
```suggestion
var skipModules = os.Getenv("GO111MODULE") != "off"
```

**Review by rjoleary — APPROVED** [review]

**Pull request merged**